hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/fsdataset/impl/FsDatasetImpl.java

```diff
@@ -2310,10 +2310,10 @@ private void invalidate(String bpid, Block[] invalidBlks, boolean async)
       throws IOException {
     final List<String> errors = new ArrayList<String>();
     for (int i = 0; i < invalidBlks.length; i++) {
-      final ReplicaInfo removing;
+      final ReplicaInfo info;
```

Comment:
Is it possible to move segment L2313~L2349 to async also? Because it includes some IO request here. # this is not blocker issue.

Reply:
How about just move the IO request out of the lock?
```
        try {
          File blockFile = new File(info.getBlockURI());
          if (blockFile != null && blockFile.getParentFile() == null) {
            errors.add("Failed to delete replica " + invalidBlks[i]
                +  ". Parent not found for block file: " + blockFile);
            continue;
          }
        } catch(IllegalArgumentException e) {
          LOG.warn("Parent directory check failed; replica {} is " +
              "not backed by a local file", info);
        }
```

Reply:
> Is it possible to move segment L2313~L2349 to async also? Because it includes some IO request here. # this is not blocker issue.

At present, it is no changed to asynchronous mainly because 
1.List<String> errors are defined in L2311, which will record the replicas that failed to delete and return IOException synchronously according to the errors list information
2.FsDatasetAsyncDiskService#deleteAsync and init ReplicaFileDeleteTask require ReplicaInfo and FsVolumeImpl etc info